internal/impl/mongodb/processor.go

```diff
@@ -228,8 +228,13 @@ func NewProcessor(conf processor.Config, mgr bundle.NewManagement) (processor.V2
 // resulting messages or a response to be sent back to the message source.
 func (m *Processor) ProcessBatch(ctx context.Context, spans []*tracing.Span, batch message.Batch) ([]message.Batch, error) {
 	writeModelsMap := map[*mongo.Collection][]mongo.WriteModel{}
-	_ = batch.Iter(func(i int, p *message.Part) error {
-		var err error
+	_ = batch.Iter(func(i int, p *message.Part) (err error) {
+		defer func() {
+			if err != nil {
+				p.ErrorSet(err)
+			}
+		}()
```

Comment:
@lucasoares this should fix the broken integration test

Reply:
Nice!